modules/core/src/lib/base-attribute.js

```diff
@@ -0,0 +1,144 @@
+    this.type = type;
+
+    if (isIndexed && !type) {
+      // If the attribute is indices, auto infer the correct type
+      // WebGL2 and WebGL1 w/ uint32 index extension support accepts Uint32Array, otherwise Uint16Array
+      this.type =
+        gl && hasFeature(gl, FEATURES.ELEMENT_INDEX_UINT32) ? GL.UNSIGNED_INT : GL.UNSIGNED_SHORT;
+    }
+
+    // Initialize the attribute descriptor, with WebGL and metadata fields
+    this.value = null;
+    this.externalBuffer = null;
+    this.buffer = null;
+    this.userData = {}; // Reserved for application
+    this.update(opts);
+
+    // Sanity - no app fields on our attributes. Use userData instead.
+    Object.seal(this);
+
+    // Check all fields and generate helpful error messages
+    this._validateAttributeDefinition();
+  }
+
+  delete() {
+    if (this.buffer) {
+      this.buffer.delete();
+      this.buffer = null;
+    }
+  }
+
+  update(opts) {
+    const {value, buffer, constant = this.constant || false} = opts;
+
+    this.constant = constant;
+
+    if (buffer) {
+      this.externalBuffer = buffer;
+      this.constant = false;
+
+      this.type = opts.type || buffer.accessor.type;
+      if (buffer.accessor.divisor !== undefined) {
+        this.divisor = buffer.accessor.divisor;
+      }
+      if (opts.divisor !== undefined) {
+        this.divisor = opts.divisor;
+      }
+    } else if (value) {
+      this.externalBuffer = null;
+      this.value = value;
+
+      if (!constant && this.gl) {
+        // Create buffer if needed
+        this.buffer =
+          this.buffer ||
+          new Buffer(
+            this.gl,
+            Object.assign({}, opts, {
+              id: this.id,
+              target: this.target,
+              type: this.type
+            })
+          );
+        this.buffer.setData({data: value});
+        this.type = this.buffer.accessor.type;
+      }
+    }
+
+    this._setAccessor(opts);
+  }
+
+  getBuffer() {
+    if (this.constant) {
+      return null;
+    }
+    return this.externalBuffer || this.buffer;
+  }
+
+  getValue() {
+    if (this.constant) {
+      return this.value;
+    }
+    const buffer = this.externalBuffer || this.buffer;
+    if (buffer) {
+      return [buffer, this];
+    }
+    return null;
+  }
+
+  // Sets all accessor props except type
+  // TODO - store on `this.accessor`
+  _setAccessor(opts) {
+    const {
+      // accessor props
+      size = this.size,
+      offset = this.offset || 0,
+      stride = this.stride || 0,
+      normalized = this.normalized || false,
+      integer = this.integer || false,
+      divisor = this.divisor || 0,
+      instanced,
+      isInstanced
+    } = opts;
+
+    this.size = size;
+    this.offset = offset;
+    this.stride = stride;
+    this.normalized = normalized;
+    this.integer = integer;
+
+    this.divisor = divisor;
+
+    if (isInstanced !== undefined) {
+      log.deprecated('Attribute.isInstanced')();
```

Comment:
Remove deprecation warnings since this is a new class?

Reply:
> Remove deprecation warnings since this is a new class?

We can certainly do that, however I think that even after this mode, we should still refactor attribute handling to align more with the `accessor` API in luma.gl (which we can now do in a non-breaking way). This will help make this system easier to understand and maintain/evolve.

I actually have another PR that starts that.  Let me put that PR up and if that doesn't look like something we should do in the near future, I will come back and remove the warnings here.